hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1053,12 +1053,24 @@ public AbfsRestOperation read(final String path,
     return op;
   }
 
-  public AbfsRestOperation deletePath(final String path, final boolean recursive, final String continuation,
+  public AbfsRestOperation deletePath(final String path, final boolean recursive,
+                                      final String continuation,
                                       TracingContext tracingContext)
           throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
-
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+
+    if (abfsConfiguration.isPaginatedDeleteEnabled() && recursive) {
+      // Change the x-ms-version to "2023-08-03" if its less than that.
+      if (xMsVersion.compareTo(AUGUST_2023_API_VERSION) < 0) {
```

Comment:
agree string comparison helps here. But I was thinking if enums can be more useful here, wherein each enum shall have an index set to it, and the comparison can be between he set-index.

Reply:
Nice suggestion...
Added a new enum class to define all the versions currently in use

Reply:
It is still string comparison. My original proposal was to assign an index to each enum, and then we compare on the basis of the index.

Reply:
This is not string comparison, this is ordinal comparison. Every enum defined has an inbuild index called ordinal which is nothing but its position in the enum set. The comparison here is comparing the position of enum in enum set.

That's why we have to add versions in chronological order.

Reply:
Got it. Nice way to achieve the required need. Thank you for the clarification!